sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ImplementSerializationMethodsCorrectly.cs

```diff
@@ -61,12 +61,18 @@ protected override void Initialize(SonarAnalysisContext context)
                 var methodDeclaration = (MethodDeclarationSyntax)c.Node;
                 var methodSymbol = c.SemanticModel.GetDeclaredSymbol(methodDeclaration);
 
-                string errorMessage;
-                if (!methodDeclaration.Identifier.IsMissing &&
-                    HasSerializationAttribute(methodSymbol) &&
-                    HasIssues(methodSymbol, out errorMessage))
+                if (methodDeclaration.Identifier.IsMissing ||
+                    !HasSerializationAttribute(methodSymbol))
                 {
-                    c.ReportDiagnostic(Diagnostic.Create(rule, methodDeclaration.Identifier.GetLocation(), errorMessage));
+                    return;
+                }
+
+                var issues = FindIssues(methodSymbol).ToList();
+                if (issues.Any())
```

Comment:
Ahaha `issues.Count > 0`. I know you will say no big deal but property access is better than method access.